### PR TITLE
aws cli session manager plugin (new formula)

### PR DESCRIPTION
**Pull request by aaronwalker**
- [ x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/aws-sessionmanager.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class AwsSessionmanager < Formula
+  desc "Session Manager Plugin for the AWS CLI"
+  homepage "https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with.html"
+  url "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/mac/sessionmanager-bundle.zip"
```
Comment (fxcoudert) [inline Formula/aws-sessionmanager.rb]: This will need a versioned URL.

File: Formula/aws-sessionmanager.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  depends_on "python"
+
+  def install
+    bin.install "bin/session-manager-plugin"
+    prefix.install_metafiles
+    prefix.install "VERSION"
+  end
+
+  test do
+    system "#{bin}/session-manager-plugin"
```
Comment (fxcoudert) [inline Formula/aws-sessionmanager.rb]: We need a test that exercises the some of the functionality of the app.

File: Formula/aws-sessionmanager.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  desc "Session Manager Plugin for the AWS CLI"
+  homepage "https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with.html"
+  url "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/mac/sessionmanager-bundle.zip"
+  version "1.0.0.0"
+  sha256 "a68e716f618937e6671b1f01b1a7c2b986457a21c0274426effba8383e15c042"
+
+  depends_on "python"
+
+  def install
+    bin.install "bin/session-manager-plugin"
```
Comment (fxcoudert) [inline Formula/aws-sessionmanager.rb]: Not sure how useful is a formula that installs a single python script…

File: Formula/aws-sessionmanager.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class AwsSessionmanager < Formula
+  desc "Session Manager Plugin for the AWS CLI"
+  homepage "https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with.html"
+  url "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/mac/sessionmanager-bundle.zip"
+  version "1.0.0.0"
+  sha256 "a68e716f618937e6671b1f01b1a7c2b986457a21c0274426effba8383e15c042"
+
+  depends_on "python"
```
Comment (fxcoudert) [inline Formula/aws-sessionmanager.rb]: Doesn't the script work with macOS system python?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/aws-sessionmanager.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+class AwsSessionmanager < Formula
+  desc "Session Manager Plugin for the AWS CLI"
+  homepage "https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with.html"
+  url "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/mac/sessionmanager-bundle.zip"
+  version "1.0.0.0"
```
Comment (fxcoudert) [inline Formula/aws-sessionmanager.rb]: Upstream says version is `1.0.0`

File: Formula/aws-sessionmanager.rb (hunk 3 above)
Comment (aaronwalker) [reply Formula/aws-sessionmanager.rb]: @fxcoudert actually it's a binary It has installer thats a python script but the formula is doing a manual install so I will remove the depends

File: Formula/aws-sessionmanager.rb (hunk 5 above)
Comment (aaronwalker) [reply Formula/aws-sessionmanager.rb]: The version string is actually stored in a `VERSION` file that states that version

File: Formula/aws-sessionmanager.rb (hunk 2 above)
Comment (aaronwalker) [reply Formula/aws-sessionmanager.rb]: this is a plugin for the aws cli so only takes a --version argument

**Pull request closed without merging**